src/palladio/SOPAssign.cpp

```diff
@@ -612,11 +635,7 @@ void SOPAssign::updatePrimitiveAttributes(GU_Detail* detail) {
 				case PRM_Type::PRM_BasicType::PRM_BASIC_FLOAT: {
 					if (parm.getMultiType() == PRM_MultiType::PRM_MULTITYPE_LIST) {
 						if (std::holds_alternative<std::vector<bool>>(it->second)) {
-							const std::optional<UT_Int32Array>& boolArray = getArrayFromMultiParm<int32>(
-							        this, parm, PRM_Type::PRM_BASIC_ORDINAL,
-							        [time](const SOPAssign* node, const PRM_Parm* parmInst) {
-								        return node->evalInt(parmInst, 0, time);
-							        });
+							const std::optional<UT_Int32Array>& boolArray = getBoolArrayFromParm(this, parm, time);
```

Comment:
this looks dangerous, you take a reference on a temporary return value. Better use return-by-value, the compiler should be able to "move" the array (return value optimziation, RVO).

also two more cases below

Reply:
fixed